hub/core/storage/shared_memory.py

```diff
@@ -0,0 +1,147 @@
+from typing import List
+from multiprocessing.shared_memory import SharedMemory
+from hub.core.storage.provider import StorageProvider
+
+
+class SharedMemoryProvider(StorageProvider):
+    """Provider class for using shared memory."""
+
+    def __init__(self, root: str = ""):
+        self.root = root
+        self.sizes = {}
+
+        # keeps the shared memory objects in memory, otherwise getitem throws warnings as the shared memory is deleted
+        self.active_shared_memories = {}
+
+    def __getitem__(self, path: str):
+        """Gets the object present at the path within the given byte range.
+
+        Example:
+            shm_provider = SharedMemoryProvider("xyz")
+            my_data = shm_provider["abc.txt"]
+
+        Args:
+            path (str): The path relative to the root of the provider.
+
+        Returns:
+            bytes: The bytes of the object present at the path.
+
+        Raises:
+            KeyError: If an object is not found at the path.
+        """
+        shared_memory = SharedMemory(name=path)
```

Comment:
Try to get existing shared memory instance from `self.active_shared_memories` first.